NEAT.js

```diff
@@ -561,140 +513,132 @@ function nodeMutate(genome) {
         gene1.weight = 1.0;
         gene1.innovation = newInnovation();
         gene1.enabled = true;
-        tableInsert(genome.genes, gene1);
+        genome.genes.push(gene1);
        
         var gene2 = copyGene(gene);
         gene2.into = genome.maxneuron;
         gene2.innovation = newInnovation();
         gene2.enabled = true;
-        tableInsert(genome.genes, gene2);
+        genome.genes.push(gene2);
 }
 
 function enableDisableMutate(genome, enable) {
-        var candidates = [];
-        for ( var i = 0; i < genome.genes; i++) {
-                if (gene.enabled == !enable) {
-                        tableInsert(candidates, gene);
-                }        	
-        }
+        var candidates = genome.genes.filter(function(gene) {
+          return gene.enabled == !enable;
+        });
+        
         if (candidates.length == 0) {
                 return;
         }       
-        var gene = candidates[randomIntFromInterval(0,candidates.length-1)];
```

Comment:
I believe it is correct, I changed the randomIntFromInterval function as well so it is a random number from [a,b), (b non inclusive)